e2e/release/src/conventional-commits-config.test.ts

```diff
@@ -0,0 +1,355 @@
+  let pkg5: string;
+
+  beforeAll(async () => {
+    newProject({
+      unsetProjectNameAndRootFormat: false,
+      packages: ['@nx/js'],
+    });
+
+    pkg1 = uniq('my-pkg-1');
+    runCLI(`generate @nx/workspace:npm-package ${pkg1}`);
+
+    pkg2 = uniq('my-pkg-2');
+    runCLI(`generate @nx/workspace:npm-package ${pkg2}`);
+
+    pkg3 = uniq('my-pkg-3');
+    runCLI(`generate @nx/workspace:npm-package ${pkg3}`);
+
+    pkg4 = uniq('my-pkg-4');
+    runCLI(`generate @nx/workspace:npm-package ${pkg4}`);
+
+    pkg5 = uniq('my-pkg-5');
+    runCLI(`generate @nx/workspace:npm-package ${pkg5}`);
+
+    // Update pkg2 to depend on pkg1
+    updateJson(`${pkg2}/package.json`, (json) => {
+      json.dependencies ??= {};
+      json.dependencies[`@proj/${pkg1}`] = '0.0.0';
+      return json;
+    });
+
+    // no git config so that the test ensures git operations happen by default
+    updateJson<NxJsonConfiguration>('nx.json', (nxJson) => {
+      nxJson.release = {
+        projectsRelationship: 'independent',
+      };
+      return nxJson;
+    });
+
+    await runCommandAsync(`git add .`);
+    await runCommandAsync(`git commit -m "chore: initial commit"`);
+    await runCommandAsync(`git tag -a ${pkg1}@0.0.1 -m "${pkg1}@0.0.1"`);
+    await runCommandAsync(`git tag -a ${pkg2}@0.0.1 -m "${pkg2}@0.0.1"`);
+    await runCommandAsync(`git tag -a ${pkg3}@0.0.1 -m "${pkg3}@0.0.1"`);
+    await runCommandAsync(`git tag -a ${pkg4}@0.0.1 -m "${pkg4}@0.0.1"`);
+    await runCommandAsync(`git tag -a ${pkg5}@0.0.1 -m "${pkg5}@0.0.1"`);
+  }, 60000);
+  afterAll(() => cleanupProject());
+
+  it('should respect custom conventional commits configuration', async () => {
+    updateJson<NxJsonConfiguration>('nx.json', (json) => {
+      json.release = {
+        ...json.release,
+        version: {
+          conventionalCommits: true,
+        },
+        changelog: {
+          projectChangelogs: {
+            renderOptions: {
+              authors: false, // do not show authors in the e2e snapshots
+            },
+          },
+        },
+        conventionalCommits: {
+          types: {
+            fix: false,
+            docs: {
+              semverBump: 'patch',
+              // no hidden property set, but the user has explicitly overridden the `docs` type, so we assume they want it to appear
+              changelog: {
+                title: 'Custom Docs Header',
+              },
+            },
+            customType: {
+              semverBump: 'minor',
+              changelog: {
+                title: 'Custom Type',
+              },
+            },
+            // unspecified semverBump will default to "patch"
+            chore: {
+              // "changelog.hidden" defaults to true, but setting changelog: false
+              // is a shortcut for setting "changelog.hidden" to false.
+              changelog: false,
+            },
+            // unspecified semverBump will default to "patch"
+            perf: {},
```

Comment:
Can we use true as a shorthand for "enabled with defaults" like we do in another places?